### ISSUE TITLE
Tapping on Quit does nothing

### ISSUE THREAD

**Issue by vikmeup** [plaid/link#185]
| Name              | Value              |
| ----------------- | ------------------ |
| Link `apiVersion` |      1.0.6              |
| Link `env`        |         development           |
| LinkKit version   |      latest              |
| iOS version       |           10.3.2         |
| iOS device        |              iPhone Se      |

### A detailed description of the steps to reproduce the issue

1) Tap on search.
2) Enter in searchBar: `Test` 
3) Press on `Quit` button

When searching for intuition returns 0 results and tapping on Quit does nothing. There is no way to handle that call back from `PLKPlaidLinkViewDelegate` delegate.

### What you expected to see?
Exit callback

### What you saw instead?
Nothing happens

### A screenshot that shows the state of the UI (if applicable)

![screen shot 2017-07-17 at 1 07 32 pm](https://user-images.githubusercontent.com/1641795/28287544-f4a05616-6af0-11e7-9131-d664b8b3e58c.png)

**Comment by vikmeup** [comment]
@pbernasconi @afh do you guys have any ETA on fix for this?

**Comment by afh** [comment]
@vikmeup Looking into it, no ETA yet, stay tuned.

**Comment by afh** [comment]
Hello @vikmeup a fix for this issue will be available with the upcoming release planned for next week.

**Comment by afh** [comment]
Hi @vikmeup this issue is fixed with the current LinkKit release [1.0.7](https://github.com/plaid/link/releases/tag/ios%2F1.0.7).

**Comment by vikmeup** [comment]
@afh currently `linkViewController(_ linkViewController: PLKPlaidLinkViewController, didExitWithError error: Error?, metadata: [String : Any]?)`

does not provide any details if it's closed from right bar button item, or quit button. Would it be possible to implement? 

Another bug is, when you open link flow, select a bank go back, then press X, it shows meta data like you had open that bank, instead of returning empty meta data, like in the first case.

**Comment by afh** [comment]
Hello @vikmeup, please see #190 regarding the details whether the flow was exited from the right bar button item or the quit button; will this work for you?

Thank you for bringing the inconsistency in the returned metadata to our attention, an improvement will be available with the next release.

**Comment by vikmeup** [comment]
I would like to also see if user pressed `Exit` button on this screen(https://github.com/plaid/link/blob/master/ios/docs/images/customization/reconnectedPane.png) or `Continue` 
At this point it still returns same message. 

It's also strange that at the last screen in the flow that `didSucceedWithPublicToken publicToken: String` not being called, even if it says that your account already been connected.

**Comment by afh** [comment]
Hi @vikmeup,

thank you for your comments.
What is your use case for distinguishing between the `Exit` and the `Continue` button?

The `linkViewController:didSucceedWithPublicToken:metadata:` delegate method is being called _after_ the `Continue` button is tapped.

**Comment by vikmeup** [comment]
> The `linkViewController:didSucceedWithPublicToken:metadata: delegate` method is being called after the Continue button is tapped.
Same method called if you press exit. It's not intuitive UX. 
I would expect to see `didCancel` call whenever user press  `Exit`. 

Since `PLKPlaidLinkViewController` derived from `UINavigationController` makes it hard to have nice flow for connecting bank account. 

My use case, when `linkViewController:didSucceedWithPublicToken:metadata:` called I would like to call `sendDeviceTokenToTheServer()` and show loading bar on top of `linkViewController`, if successfully completed `linkViewController.dismiss()` otherwise show `UIAlertController` on top of  `linkViewController`.
Any UX suggestions to make this happen? 

Currently flow would be:
`present(linkViewController)` -> `dismissed linkViewController with a token (exit button confusing for that congrats screen) ` -> `push or present something(loading) to save token?`. Flaky UX in this case.

Do you have any examples of other apps implementing Plaid SDK into their flow? would be curious to se those.

Thank you!

**Comment by afh** [comment]
Hi @vikmeup 

here's a rough example that adds an overlay view showing an activity indicator while the application is busy uploading the public token to a server.

If LinkKit is shown on iPad in a modal view the cornerRadius of the overlay view needs to be set to match the modal appearance, additional device rotation handling to update the overlay is also necessary.

```swift
    func didTapDeviceTokenUploadOverlay() {
        // Handle user interaction on overlay view here
        NSLog("didTapDeviceTokenUploadOverlay")
    }
    
    func linkViewController(_ linkViewController: PLKPlaidLinkViewController, didSucceedWithPublicToken publicToken: String, metadata: [String : Any]?) {
        let offset = linkViewController.view.convert(self.view.frame, to: nil).origin
        let overlayView = UIControl(frame: linkViewController.view.frame.offsetBy(dx: offset.x, dy: offset.y))
        overlayView.backgroundColor = UIColor.black.withAlphaComponent(0.3)
        overlayView.addTarget(self, action: #selector(ViewController.didTapDeviceTokenUploadOverlay), for: .touchUpInside)
        
        let loadingView = UIView(frame: CGRect(x: 0, y: 0, width: 100, height: 100))
        loadingView.backgroundColor = UIColor.black.withAlphaComponent(0.8)
        loadingView.layer.cornerRadius = 8
        loadingView.center = linkViewController.view.center
        overlayView.addSubview(loadingView)
        
        let activityView = UIActivityIndicatorView(activityIndicatorStyle: .white)
        activityView.center = CGPoint(x: loadingView.bounds.midX, y: loadingView.bounds.midY)
        activityView.startAnimating()
        loadingView.addSubview(activityView)
        
        linkViewController.view.isUserInteractionEnabled = false
        linkViewController.view.window?.addSubview(overlayView)
        
        let sendDeviceTokenToTheServer = { (publicToken: String, completion: @escaping ()->()) in
            let when = DispatchTime.now() + 5 // Simulate publicToken upload for 5 seconds
            DispatchQueue.main.asyncAfter(deadline: when, execute: completion)
        }
        
        NSLog("Successfully linked account!\npublicToken: \(publicToken)\nmetadata: \(metadata ?? [:])")
        sendDeviceTokenToTheServer(publicToken) {
            NSLog("Public token upload successful")
            overlayView.removeFromSuperview()
            self.dismiss(animated: true) {
                self.handleSuccessWithToken(publicToken, metadata: metadata)
            }
        }
    }
```

**Comment by vikmeup** [comment]
@afh the last thing I have left to distinguish between `X` and `Continue` (screenshot below) button when pressed, because they both call `linkViewController:didSucceedWithPublicToken:metadata:` with the same result. I would like to handle `X` for dismissing the flow, and `Continue` for saving public token.

Do you think it will get into next release?

![simulator screen shot - iphone se - 2017-08-24 at 13 13 48](https://user-images.githubusercontent.com/1641795/29686520-1cda6af4-88ce-11e7-9ca4-53c14b0f0de8.png)

**Comment by afh** [comment]
Hi @vikmeup at this point the account has successfully been linked with Plaid and an item was created, so `×` at this point means the same as `Continue` rather than "cancel" or "abort".

While we are considering changes to this pane, it seems rather unlikely that they include a way to distinguish between the `×` and `Continue` button.

**Comment by vikmeup** [comment]
@afh any change you can add ability to control that flow? UI wise it's confusing for users to understand that.

**Comment by afh** [comment]
Hi @vikmeup thanks for the feedback that it potentially confuses users, as stated before we are considering changes to the result pane to improve the experience for users.

**Comment by afh** [comment]
Hi @vikmeup the newly released of version of LinkKit 1.1.0 removes the `×`-button from the result pane addressing the feedback you had.
